packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,3235 @@
+  final EdgeInsets? padding;
+
+  // The padding around the edges of a submenu. Passed in from the MenuBarMenu
+  // so that it can be given during registration with the controller.
+  final EdgeInsets? _menuPadding;
+
+  /// The text style for the text in this menu bar item.
+  ///
+  /// May be overridden inside of [labelWidget], if supplied.
+  ///
+  /// Defaults to the value of [MenuBarThemeData.itemTextStyle] if not set.
+  final MaterialStateProperty<TextStyle?>? textStyle;
+
+  /// The shape of this menu bar item.
+  ///
+  /// Defaults to the value of [MenuBarThemeData.itemShape] if not set.
+  final MaterialStateProperty<OutlinedBorder?>? shape;
+
+  // Indicates that this is a button for a submenu, not just a regular item.
+  final bool _hasMenu;
+
+  @override
+  State<MenuBarItem> createState() => _MenuBarItemState();
+
+  @override
+  Iterable<Map<String, Object?>> toChannelRepresentation(PlatformMenuDelegate delegate,
+      {required int Function(MenuItem) getId}) {
+    return <Map<String, Object?>>[PlatformMenuItem.serialize(this, delegate, getId)];
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(FlagProperty('enabled', value: onSelected != null || onSelectedIntent != null, ifFalse: 'DISABLED'));
+    properties.add(DiagnosticsProperty<FocusNode>('focusNode', focusNode, defaultValue: null));
+    properties.add(DiagnosticsProperty<Widget>('leadingIcon', leadingIcon, defaultValue: null));
+    properties.add(StringProperty('label', label));
+    properties.add(DiagnosticsProperty<Widget>('trailingIcon', trailingIcon, defaultValue: null));
+    properties.add(StringProperty('semanticLabel', semanticLabel, defaultValue: null));
+    properties.add(DiagnosticsProperty<EdgeInsets?>('padding', padding, defaultValue: null));
+    properties.add(
+        DiagnosticsProperty<MaterialStateProperty<Color?>>('backgroundColor', backgroundColor, defaultValue: null));
+    properties.add(
+        DiagnosticsProperty<MaterialStateProperty<Color?>>('foregroundColor', foregroundColor, defaultValue: null));
+    properties
+        .add(DiagnosticsProperty<MaterialStateProperty<Color?>>('overlayColor', overlayColor, defaultValue: null));
+    properties.add(DiagnosticsProperty<MaterialStateProperty<TextStyle?>>('textStyle', textStyle, defaultValue: null));
+  }
+}
+
+class _MenuBarItemState extends State<MenuBarItem> {
+  _MenuNode? menu;
+  int menuSerial = 0;
+  late _MenuBarController controller;
+  bool registered = false;
+  Timer? hoverTimer;
+  FocusNode get focusNode => widget.focusNode ?? _focusNode!;
+  FocusNode? _focusNode;
+
+  bool get enabled {
+    return (widget.onSelected != null || widget.onSelectedIntent != null) && controller.enabled;
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _focusNode = FocusNode(debugLabel: 'MenuBarItem(${widget.label})');
```

Comment:
to match didUpdateWidget - only do this when widget.focusNode is null?

Reply:
Oh, good point.